cl/citations/management/commands/cl_find_citations_recap.py

```diff
@@ -0,0 +1,180 @@
+                    options.get("start_id") is None,
+                    options.get("end_id") is None,
+                    options.get("filed_after") is None,
+                    options.get("filed_before") is None,
+                    options.get("modified_after") is None,
+                    options.get("all") is False,
+                ]
+            )
+            if both_list_and_endpoints or no_option:
+                raise CommandError(
+                    "Please specify either a list of documents, a "
+                    "range of ids, a range of dates, or "
+                    "everything."
+                )
+
+            self.index = options["index"]
+            query = RECAPDocument.objects.all().order_by("pk")
+            if options.get("doc_id"):
+                query = query.filter(pk__in=options["doc_id"])
+            if options.get("end_id"):
+                query = query.filter(pk__lte=options["end_id"])
+            if options.get("start_id"):
+                query = query.filter(pk__gte=options["start_id"])
+
+            if options.get("filed_after"):
+                query = query.filter(
+                    cluster__date_filed__gte=options["filed_after"]
+                )
+
+            if options.get("filed_before"):
+                query = query.filter(
+                    cluster__date_filed__lte=options["filed_before"]
+                )
+
+            if options.get("modified_after"):
+                query = query.filter(
+                    date_modified__gte=options["modified_after"]
+                )
+
+            if options.get("all"):
+                query = RECAPDocument.objects.all()
+
+            self.count = query.count()
+            self.average_per_s = 0.0
+            self.timings: List[float] = []
+
+            document_pks = query.values_list("pk", flat=True).iterator()
+            queue = cast(str, options["queue"])
+            self.update_documents(document_pks, queue)
+
+            def log_progreess(self, processed_count: int, last_pk: int):
+                if processed_count % 1000 == 1:
+                    self.t1 = time.time()
+                if processed_count % 1000 == 0:
+                    self.t2 = time.time()
+                    self.timings.append(self.t2 - self.t1)
+                    self.average_per_s = (
+                        1000 / sum(self.timings) / float(len(self.timings))
+                    )
+
+                template = (
+                    "\rProcessing items in Celery queue: {:.0%} ({}/{}, "
+                    "{:.1f}/s, Last id: {})"
+                )
+
+                sys.stdout.write(
+                    template.format(
+                        float(processed_count) / self.count,  # Percent
+                        processed_count,
+                        self.count,
+                        self.average_per_s,
+                        last_pk,
+                    )
+                )
+
+                sys.stdout.flush()
```

Comment:
I guess, also, I'm sorry to say, I'm realizing this isn't DRY, and I'm wondering how hard it would be the factor out this function. Maybe it's not worth it, but it's a thought.

Reply:
It wouldn't be that hard to put it in the base Class, though it's unclear whether it belongs in a Class called `VerboseCommand`. Aside from these 2 very similar commands, the others in `citations` do not use the same logging. I'm unsure what the logging looks like throughout the rest of the system, so I guess I offer limited help here.

Another option to consider is merging the 2 commands for finding citations in stored documents into one.

Reply:
Eh. Let's leave it be.